engine/system/core/src/main/java/it/unibz/inf/ontop/answering/connection/impl/QuestStatement.java

```diff
@@ -224,8 +227,10 @@ private SimpleGraphResultSet executeDescribeQuery(DescribeQuery inputQuery, Immu
 					describeResultSet = set;
 				} else if (set != null) {
 					// 2nd and manyth times execute, but collect result into one object
-					while (set.hasNext())
-						describeResultSet.addNewResult(set.next());
+					RDFFactCloseableIterator iterator = set.iterator();
```

Comment:
Will have to use the interface `OntopCloseableIterator<RDFFact, OntopConnectionException>`